src/Waargonaut/Encode.hs

```diff
@@ -91,180 +96,190 @@ import qualified Data.Map                   as Map
 import           Data.Text                  (Text)
 
 import           Waargonaut                 (waargonautBuilder)
-import           Waargonaut.Types           (AsJType (..), Json,
-                                             MapLikeObj (..), WS, textToJString,
-                                             wsRemover, _JNumberInt)
+import           Waargonaut.Types           (AsJType (..), JAssoc (..), JObject,
+                                             Json, MapLikeObj (..), WS,
+                                             textToJString, wsRemover,
+                                             _JNumberInt)
 
 -- |
 -- Define an "encoder" as a function from some @a@ to some 'Json' with the
 -- allowance for some context @f@.
 --
-newtype Encoder' f a = Encoder'
+newtype Encoder f a = Encoder
```

Comment:
I think it would be really useful to add a function of type `Applicative f => Encoder' a -> Encode f a` for lifting from a normal encoder to a fancy one. I think it's likely if I were using `f` that I would have one or two subcomponents that actually used f, and the rest would just need to be lifted. Parametricity would do this for us if we started with the non-prime primitives, but still I think this would be a nice thing to have.